### ISSUE TITLE
Akka cluster splitbrain on startup

### ISSUE THREAD

**Issue by brianwawok** [akka/akka#18554]
I was sent here from the user_group by @patriknw 

https://groups.google.com/forum/#!topic/akka-user/5RmljydLk3g

I am using AKKA 2.3.13

```
cluster {
  min-nr-of-members = 2
  seed-nodes = [
    "akka.tcp://System1@ip-197:9010",
    "akka.tcp://System1@ip-85:9010",
    "akka.tcp://System1@ip-86:9010"]
  auto-down-unreachable-after = off
  failure-detector {
    threshold = 16.0
    acceptable-heartbeat-pause = 10 s
    expected-response-after = 12 s
  }
```

I use the cluster singleton to split up work. 1 guy becomes the master, and the other 2 become workers.

Cool, 75% of the time it works.  25% of the time - it fails on startup. For some reason the nodes form 2 small clusters, one with 2 nodes, and one with only 1 node.  Once the cluster is up all is well.

My deployment procedure is:

1) Kill -9 all nodes
2) Wait 5 seconds
3) Start all 3 nodes one after another (few ms lag between them).

As you can see, the 197 IP is listed first.. so I guess that means everyone should converge on him. On my last event of this, I have 1 cluster of 197 and 85, with 85 elected the singleton.. and a 2nd cluster with 86 as the only member.

Crux of the problem is these log lines conflict on who the singleton is

```
Singleton identified: akka.tcp://MySystem@IP.85:9010/user/singleton/overmaster
Singleton identified: akka://MySystem/user/singleton/overmaster
Singleton identified: akka://MySystem/user/singleton/overmaster
```

1 and 2 agree that 2 is the singleton. 3 thinks that he is the singleton.

Log1: https://gist.github.com/brianwawok/d895d7f22891ec4b07c3
Log2: https://gist.github.com/brianwawok/3593efe4be78a5c70cb2
Log3: https://gist.github.com/brianwawok/3b71ff75a98ffa983c41

**Comment by patriknw** [comment]
@brianwawok this is a bug. Thanks for reporting. Until we have fixed it you can work around it by using a delay (such as 5 seconds) between starting the nodes. Another way to mitigate the risk of this happening is to increase the `akka.cluster.leader-actions-interval` to let's say 4 seconds.

**Comment by brianwawok** [comment]
it does look like my 3rd server has 3 or 4 second lead on the time, not sure if small clock skew is helping this along...

**Comment by patriknw** [comment]
Yes, it is very clear from the logs what is going on. Clock skew is not the problem.

The "leader" is simply the lowest address, with some additional rules. So it's not a strict leader election. node-2 should be the leader in this 3 node cluster, but node-2 thinks that it is the leader for a short while (1 second), before it knows about node-2. During this period both node-2 and node-3 moves nodes to Up, but node-3 doesn't know about node-2 yet, so therefore the upNumber becomes different. node-2 things node-2 is oldest but node-3 thinks node-3 is oldest. Yeah, interesting problem.

**Comment by brianwawok** [comment]
FWIW node 1 has a different and higher third octet (but in either case, node 2 seems to be "lowest" form a math.min point of view.

This is interesting. Only a simple three node cluster here, I am hoping to scale this cluster to 10, 20, 30... seems like it may be even more complex as I add more nodes.

Trying 5 seconds between starts, should help with any race conditions I suppose?

**Comment by patriknw** [comment]
Yes, it's really just a problem for the very first nodes, so when you have started a few with delays you can start many at the same time. We should of course fix this caveat.

**Comment by patriknw** [comment]
@brianwawok another workaround that should work without any delays is to use the lowest address as the first element in the seed-nodes list, i.e. akka.tcp://MySystem@IP.197:9010 in your example. By lowest I mean alphanumerical sorting of the hostname/ip string, followed by lowest port for equal hostname.

**Comment by brianwawok** [comment]
@patriknw as a random outsider stepping in.. logic would say if I said my seed nodes are:  5, 1, 4.. they would be tried in that order (not alphabetical order 1,4,5). So my identically shared config across nodes SHOULD have worked. 

is perhaps the alphabetical sorting of nodes 1 step too clever?  I find it more common that a new person would give the same seed node list to all nodes over a new person alphabetically sorting IPs (what???)

Though I guess if the bug is fixed the point is a bit moot...

**Comment by patriknw** [comment]
It is the first seed node that is special, and important. The order of the other seed nodes are not important. In hindsight we should have made the first seed node a separate explicit configuration property.

I know how to fix this issue, I just need to find the time to type it in with some tests.

**Comment by patriknw** [comment]
@brianwawok fix merged to master, will be included in upcoming 2.4.1

**Comment by brianwawok** [comment]
@patriknw awesome!
